pkg/db/proxy.go

```diff
@@ -69,27 +69,32 @@ type remoteImpl struct {
 	localDB       DB
 	whitelist     map[string]bool
 	postURL       string
-	log           *util.Logger
+	logger        log.Logger
 	wlHistory     map[string]*lru.ARCCache
 	isRemote      bool
 	rwLock        sync.RWMutex
 }
 
-func OpenRemote(cfg *config.Config, localDB DB) (DataServerProxy, error) {
-	return open(cfg, localDB, true)
+func OpenRemote(logger log.Logger, cfg *config.Config, localDB DB) (DataServerProxy, error) {
+	return open(logger, cfg, localDB, true)
 }
 
-func OpenLocal(cfg *config.Config, localDB DB) (DataServerProxy, error) {
-	return open(cfg, localDB, false)
+func OpenLocal(logger log.Logger, cfg *config.Config, localDB DB) (DataServerProxy, error) {
+	return open(logger, cfg, localDB, false)
 }
 
 // OpenRemoteDB establishes a proxy to a remote data server.
-func open(cfg *config.Config, localDB DB, isRemote bool) (DataServerProxy, error) {
-	rdbLog = util.NewLogger("db", "RemoteDBProxy")
+func open(logger log.Logger, cfg *config.Config, localDB DB, isRemote bool) (DataServerProxy, error) {
+	filterLog, err := logging.ApplyFilter(*cfg, ComponentName, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "applying filter to logger")
+	}
+
+	rdbLog := log.With(filterLog, "component", "RemoteDBProxy")
```

Comment:
```suggestion
	rdbLog := log.With(filterLog, "component", ComponentName)
```
please make all component names constants.